app/lib/admin/models.dart

```diff
@@ -79,6 +88,34 @@ class ModerationCase extends db.ExpandoModel<String> {
     id = caseId;
     opened = clock.now().toUtc();
   }
+
+  ModerationActionLog getActionLog() {
+    if (actionLogField == null) {
+      return ModerationActionLog(entries: []);
+    }
+    return ModerationActionLog.fromJson(
+        json.decode(actionLogField!) as Map<String, Object?>);
+  }
+
+  void setActionLog(ModerationActionLog value) {
+    actionLogField = json.encode(value.toJson());
+  }
+
+  /// Adds a new entry after deduplication (skipping the current
+  /// [subject]+[isModerated] pair if an identical exists as the last state).
+  void addActionLogEntryWithDeduplication(String subject, bool isModerated) {
+    final log = getActionLog();
+    final lastEntry = log.entries.where((e) => e.subject == subject).lastOrNull;
+    if (lastEntry != null && lastEntry.isModerated == isModerated) {
+      // duplicate entry found, skip inserting
```

Comment:
Only, if date-time is within 10 minutes? We just want retried request to not trigger new entries, right?

In that case, I think we should clearly state why in a comment.

Reply:
I'm not sure it is only 10 minutes. It may be somebody starts a restrictive admin action, which is partially completed, as not everything happens in the same transaction. It may be after an hour that the action is repeated.

Reply:
let's make it 10-15 minutes, if someone really does reapply it, what is the harm of having two entries?

We kind of want to have the date-time from both.

I think it's mostly worth deduplicating in cases where retries redo the operation.

-----------------

Actually, unless harmful I suggest we consider not deduplicating at all. If we need to do undo a restrictive action due to appeal, that will happen in a different `ModerationCase`.